src/backend/routers/routes.test.ts

```diff
@@ -0,0 +1,61 @@
+import test from "ava";
+import { trpcRouter } from "@/backend/routers/_app";
+import { createContext } from "@/backend/context";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { TRPCError } from "@trpc/server";
+
+test("All API endpoints are auth guarded", async (t) => {
+  // Create a more complete mock of req and res
+  const mockReq = {
+    headers: {},
+    cookies: {},
+  } as unknown as NextApiRequest;
+
+  const mockRes = {
+    getHeader: () => undefined,
+    setCookie: () => void 0,
+    setHeader: () => void 0,
+  } as unknown as NextApiResponse;
+
+  // Create a mock context with no authentication
+  const ctx = await createContext({
+    req: mockReq,
+    res: mockRes,
+  });
+  // Define an exclude list for certain router/procedures
+  const excludeList = ["public.healthCheck"];
+
+  // Create a caller with this context
+  const caller = trpcRouter.createCaller(ctx);
+
+  // Get all procedure names
+  const procedureNames = Object.keys(trpcRouter._def.procedures);
+
+  for (const fullProcedureName of procedureNames) {
+    if (excludeList.includes(fullProcedureName)) {
+      continue;
+    }
+    const [routerName, procedureName] = fullProcedureName.split(".");
+
+    try {
+      await (
+        caller[routerName as keyof typeof caller] as Record<
+          string,
+          (...args: unknown[]) => Promise<unknown>
```

Comment:
I'm curious if there's a reason to use '...args: unknown[]' instead of just leaving the parameters empty?

Reply:
@KCCPMG good catch! simplified in ebe1f54ae67a3657566bda37ea54b70a649f8bb8